statistics/table.go

```diff
@@ -943,6 +950,22 @@ func (coll *HistColl) GetIndexAvgRowSize(ctx sessionctx.Context, cols []*express
 	return
 }
 
+// GetValidColumn returns the column stats if it exists and is valid. Otherwise it returns nil.
+// Also, it tracks the used(needed) column stats in sc.ColStatsUsage.
+func (coll *HistColl) GetValidColumn(sc *stmtctx.StatementContext, colID int64, collPseudo bool) *Column {
+	if coll.UniqueID2TblColID != nil {
+		if tblColID, ok := coll.UniqueID2TblColID[colID]; ok {
+			// track the column stats usage
+			// TODO: check the assumption that all the column stats usage is accessed by UniqueID
+			sc.ColStatsUsage[tblColID] = time.Now()
```

Comment:
It might not be safe here? When there's self-join in the SQL, we may access the stats of one table concurrently?